src/compiler/binder.ts

```diff
@@ -1751,4 +1791,440 @@ namespace ts {
             implicitLabels = [];
         }
     }
+
+    /**
+     * Computes the transform flags for a node, given the transform flags of its subtree
+     *
+     * @param node The node to analyze
+     * @param subtreeFlags Transform flags computed for this node's subtree
+     */
+    export function computeTransformFlagsForNode(node: Node, subtreeFlags: TransformFlags) {
+        // Ambient nodes are TypeScript syntax and the flags of their subtree are ignored.
+        if (node.flags & NodeFlags.Ambient) {
+            return (node.transformFlags = TransformFlags.AssertTypeScript)
+                & ~(node.excludeTransformFlags = TransformFlags.NodeExcludes);

```

Comment:
Yes, `~` gets the ones-complement of the operand. The result here is the `transformFlags` of the node, excluding `excludeTransformFlags` of the node.